lib/web_ui/lib/src/engine/window.dart

```diff
@@ -72,11 +73,32 @@ base class EngineFlutterView implements ui.FlutterView {
 
   final ViewConfiguration _viewConfiguration = const ViewConfiguration();
 
+  /// Whether this [EngineFlutterView] has been disposed or not.
+  bool isDisposed = false;
+
+  /// Disposes of the [EngineFlutterView] instance and undoes all of its DOM
+  /// tree and any event listeners.
+  @mustCallSuper
+  void dispose() {
+    isDisposed = true;
+    platformDispatcher.unregisterView(this);
+    dimensionsProvider.close();
+    dom.rootElement.remove();
+    // TODO(harryterkelsen): What should we do about this in multi-view?
+    renderer.clearFragmentProgramCache();
+  }
+
   @override
-  void render(ui.Scene scene) => platformDispatcher.render(scene, this);
+  void render(ui.Scene scene) {
+    assert(!isDisposed, 'Trying to render a disposed EngineFlutterView.');
```

Comment:
Not 100% sure, but I think this should not be an assert, just a noop.

(Also since this is being called from `views[id].render()`, it's likely that this assert will only be exercised in test code, right?)

Reply:
> (Also since this is being called from `views[id].render()`, it's likely that this assert will only be exercised in test code, right?)

In theory, yes. In practice, the assert here serves as a guard against bugs in our logic if we somehow try to render to a disposed view.

If the framework tries to render to a disposed view, then the platform dispatcher (or `renderer`) should do `viewData[id]?.render(scene)` (which would fail to find the view, and it'll be a noop). I don't think we should allow calling `render` directly on a disposed view.